src/files/FilesPage.js

```diff
@@ -208,7 +266,46 @@ const FilesPage = ({
         onAddByPath={(files) => showModal(ADD_BY_PATH, files)}
         onNewFolder={(files) => showModal(NEW_FOLDER, files)}
         onCliTutorMode={() => showModal(CLI_TUTOR_MODE)}
-        handleContextMenu={(...args) => handleContextMenu(...args, true)} />
+        handleContextMenu={(...args) => handleContextMenu(...args, true)}
+      />
+
+      {(files && files.type !== 'file') && <div className="flex items-center justify-between">
+        <div>
+          {viewMode === 'grid' && files?.content?.length > 0
+            ? (
+                <Checkbox
+                  className='pv3 pl3 pr1 bg-white flex-none'
+                  onChange={(checked) => {
+                    if (checked) {
+                      setSelected(files.content.map(f => f.name))
+                    } else {
+                      setSelected([])
+                    }
+                  }}
+                  checked={files?.content?.length > 0 && selected.length === files.content.length}
+                  label={<span className='fw5 f6'>{t('selectAllEntries')}</span>}
+                />
+              )
+            : null
+          }
+        </div>
+        <div className="flex items-center justify-end ml3">
+          <button
+            className={`pa2 pointer ${viewMode === 'list' ? 'selected-item' : 'gray'}`}
+            onClick={() => setViewMode('list')}
+            title={t('viewList')}
```

Comment:
Translation file seems to be missing `viewList` and `viewGrid` keys.

(If you don't have bandwidth, just add it to english one –  other languages will be translated via https://app.transifex.com/ipfs/ipfs-webui/dashboard/  once this PR lands in main branch)